java/adapter/avro/src/main/java/org/apache/arrow/consumers/NullablePrimitiveTypeConsumer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.consumers;
+
+import java.io.IOException;
+
+import org.apache.avro.io.Decoder;
+
+/**
+ * Consumer holds a primitive consumer, could consume nullable values from avro decoder.
+ * Write data via writer of delegate consumer.
+ */
+public class NullablePrimitiveTypeConsumer implements Consumer {
```

Comment:
Would this not work for complex types?

Reply:
I think it won't work for complex types, for unions type like ["string", "int", "null"] I think should add a consumer which holds something like Map<index, delegate>.
This PR only aims for nullable primitive type represented by union type.

Reply:
Agreed, by complex I mean something like [Struct, Null] or [List, Null].  I think a better name would be NullableTypeConsumer?

Reply:
Ah, right, fixed.